core/io/image.cpp

```diff
@@ -2611,23 +2611,25 @@ Image::AlphaMode Image::detect_alpha() const {
 }
 
 Error Image::load(const String &p_path) {
+	String path = ResourceUID::ensure_path(p_path);
 #ifdef DEBUG_ENABLED
-	if (p_path.begins_with("res://") && ResourceLoader::exists(p_path)) {
```

Comment:
This function is kind of tricky, but I think if you supply an UID it should just not work.

Reply:
See the linked issue.
The expectation is different apparently.